contract-verification/src/main/scala/sigmastate/verification/SigmaDsl/api/sigma/SigmaDsl.scala

```diff
@@ -0,0 +1,724 @@
+}
+
+/** Represents data of the block header available in Sigma propositions.
+  * @since 2.0
+  */
+@scalan.Liftable
+@library
+trait Header {
+  /** Bytes representation of ModifierId of this Header */
+  def id: Coll[Byte]
+
+  /** Block version, to be increased on every soft and hardfork. */
+  def version: Byte
+
+  /** Bytes representation of ModifierId of the parent block */
+  def parentId: Coll[Byte] //
+
+  /** Hash of ADProofs for transactions in a block */
+  def ADProofsRoot: Coll[Byte] // Digest32. Can we build AvlTree out of it?
+
+  /** AvlTree) of a state after block application */
+  def stateRoot: AvlTree
+
+  /** Root hash (for a Merkle tree) of transactions in a block. */
+  def transactionsRoot: Coll[Byte]  // Digest32
+
+  /** Block timestamp (in milliseconds since beginning of Unix Epoch) */
+  def timestamp: Long
+
+  /** Current difficulty in a compressed view.
+    * NOTE: actually it is unsigned Int*/
+  def nBits: Long
+
+  /** Block height */
+  def height: Int
+
+  /** Root hash of extension section */
+  def extensionRoot: Coll[Byte] // Digest32
+
+  /** Miner public key. Should be used to collect block rewards.
+    * Part of Autolykos solution. */
+  def minerPk: GroupElement
+
+  /** One-time public key. Prevents revealing of miners secret. */
+  def powOnetimePk: GroupElement
+
+  /** nonce */
+  def powNonce: Coll[Byte]
+
+  /** Distance between pseudo-random number, corresponding to nonce `powNonce` and a secret,
+    * corresponding to `minerPk`. The lower `powDistance` is, the harder it was to find this solution. */
+  def powDistance: BigInt
+
+  def votes: Coll[Byte] //3 bytes
+}
+
+/** Represents data available in Sigma language using `CONTEXT` global variable*/
+@scalan.Liftable
+@library
+trait Context {
+  def builder: SigmaDslBuilder
+
+  /** A collection of outputs of the current transaction. */
+  def OUTPUTS: Coll[Box]
+
+  /** A collection of inputs of the current transaction, the transaction where selfBox is one of the inputs. */
+  def INPUTS: Coll[Box]
+
+  /** A collection of inputs of the current transaction that will not be spent. */
+  def dataInputs: Coll[Box]
+
+  /** Height (block number) of the block which is currently being validated. */
+  def HEIGHT: Int
+
+  /** Box whose proposition is being currently executing */
+  def SELF: Box
+
+  /** Zero based index in `inputs` of `selfBox`. -1 if self box is not in the INPUTS collection. */
+  def selfBoxIndex: Int
+
+  /** Authenticated dynamic dictionary digest representing Utxo state before current state. */
+  def LastBlockUtxoRootHash: AvlTree
+
+  /** A fixed number of last block headers in descending order (first header is the newest one)
+    * @since 2.0
+    */
+  def headers: Coll[Header]
+
+  /**
+    * @since 2.0
+    */
+  def preHeader: PreHeader
+
+  def minerPubKey: Coll[Byte]
+  def getVar[T](id: Byte)(implicit cT: RType[T]): Option[T]
+  def vars: Coll[AnyValue]
+}
+
+@scalan.Liftable
+@library
+trait SigmaContract {
```

Comment:
>The only reason why SigmaContract is not removed is because ContractSyntax is used ErgoDsl and it is used in Specs.
>So, if it is easier, you can use basic SigmaDsl traits and operations directly to generate scalaFunc.

We need to anchor our predefined functions somewhere. I mean in ErgoScala we need to be able to call `atLeast(...)` in the contract. Do you want to call predefined functions directly on `SigmaDslBuilder`? How do we pass it into the contract? I like the idea of a contract extending `SigmaContract` trait and have predefined functions.